### PR TITLE
adds doc on configuring quay-enterprise with tectonic-identity

### PR DESCRIPTION
**Pull request by kbrwn**
This doc has been tested by myself and another support engineer.

### CODE REVIEW

File: quay-tectonic-identity.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Quay Enterprise OIDC Auth with Tectonic Identity 
+
```
Comment (zbwright) [inline quay-tectonic-identity.md]: need an introductory sentence here.

File: quay-tectonic-identity.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+## Export certificate information from cluster for TLS verification of Tectonic Identity: 
+
+```
+kubectl -n tectonic-system get secret tectonic-ca-cert-secret -o json | jq -r '.data[]' | base64 -d >> tectonic-ca-ingress.crt
+kubectl -n tectonic-system get secret tectonic-ingress-tls-secret -o json | jq -r '.data["tls.crt"]' | base64 -d >> tectonic-ca-ingress.crt
+```
```
Comment (zbwright) [inline quay-tectonic-identity.md]: add blank line after ```

File: quay-tectonic-identity.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+## Upload Cert to QE
+
+Upload the `tectonic-ca-ingress.crt` certificate to QE via the superuser panel or follow the [documentation](https://coreos.com/quay-enterprise/docs/latest/insert-custom-cert.html) upload manually.
```
Comment (zbwright) [inline quay-tectonic-identity.md]: ... to Quay through the superuser panel, or follow [Adding TLS Certificates to the Quay Enterprise Container][add-certs] to upload manually.

File: quay-tectonic-identity.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+Save the changes to the `tectonic-identity` configmap and issue a `kubectl patch` to update the deployment: 
+
+```
+kubectl --namespace tectonic-system patch deployment tectonic-identity \
+    --patch "{\"spec\":{\"template\":{\"metadata\":{\"annotations\":{\"date\":\"`date +'%s'`\"}}}}}"
+```
+
+Navigate back to the Quay Enterprise superuser panel click "Save Configuration Changes", then "Save Configuration, and finally restart the Quay Enterprise container with a `docker restart` or `kubectl delete pod <quay-enterprise>`. 
+
+
```
Comment (zbwright) [inline quay-tectonic-identity.md]: [add-certs]: https://coreos.com/quay-enterprise/docs/latest/insert-custom-cert.html

File: quay-tectonic-identity.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+```
+  - id: quay-enterprise
+    name: Quay Enterprise
+    secret: rEpLaCeThIs
+    redirectURIs:
+      - ''
+```
+
+`secret:` is a user generated string that will later be passed to Quay Enterprise. Use `date | md5sum` or `openssl rand -base64 15` to generate a random string.
```
Comment (zbwright) [inline quay-tectonic-identity.md]: `secret`: a user generated ...

File: quay-tectonic-identity.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+```
+  - id: quay-enterprise
+    name: Quay Enterprise
+    secret: rEpLaCeThIs
+    redirectURIs:
+      - ''
+```
+
+`secret:` is a user generated string that will later be passed to Quay Enterprise. Use `date | md5sum` or `openssl rand -base64 15` to generate a random string. 
+`redirectURIs:` URLs that must be passed to the tectonic-identity configmap. These are provided by Quay Enterprise in the next step.
```
Comment (zbwright) [inline quay-tectonic-identity.md]: `redirectURIs`:

File: quay-tectonic-identity.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+Leave this terminal open for use later
```
Comment (zbwright) [inline quay-tectonic-identity.md]: ... period.

File: quay-tectonic-identity.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+## Add an OIDC Provider for Tectonic Identity
+
+From Quay Enterprise, navigate to the superuser panel, then to External Authorization, click the button: `Add OIDC Provider`, and type `tectonicidentity` in the prompt.
+
+`ODIC Server:` is the `issuer:` parameter from the tectonic-identity configmap with a forwardslash at the end: 'https://tectonic.example.com/identity/'
```
Comment (zbwright) [inline quay-tectonic-identity.md]: `OIDC`: the `issuer` parameter ...at the end. For example: `https ...`.

File: quay-tectonic-identity.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+`Client ID`: `quay-enterprise`
```
Comment (zbwright) [inline quay-tectonic-identity.md]: `Client ID`:

File: quay-tectonic-identity.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+`Client Secret:` String passed as `secret` to tectonic-identity configmap
```
Comment (zbwright) [inline quay-tectonic-identity.md]: `Client Secret`: string passed ... period.

File: quay-tectonic-identity.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+`Service Name:`	will be displayed on the login page. Suggested Value: `Tectonic Identity`
+
+## Configure Redirect URIs
+
+Navigate back to editing the `tectonic-identity` configmap. Add the provided CallbackURLs as redirectURIs:
```
Comment (zbwright) [inline quay-tectonic-identity.md]: navigate back to where? navigate back to the termiinal

File: quay-tectonic-identity.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+```
+  - id: quay-enterprise
+    name: Quay Enterprise
+    secret: rEpLaCeThIs
+    redirectURIs:
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback'
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback/attach'
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback/cli'
+```
+
```
Comment (zbwright) [inline quay-tectonic-identity.md]: extra blank line.

File: quay-tectonic-identity.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+
+Save the changes to the `tectonic-identity` configmap and issue a `kubectl patch` to update the deployment: 
+
+```
+kubectl --namespace tectonic-system patch deployment tectonic-identity \
+    --patch "{\"spec\":{\"template\":{\"metadata\":{\"annotations\":{\"date\":\"`date +'%s'`\"}}}}}"
+```
+
+Navigate back to the Quay Enterprise superuser panel click "Save Configuration Changes", then "Save Configuration, and finally restart the Quay Enterprise container with a `docker restart` or `kubectl delete pod <quay-enterprise>`.
```
Comment (zbwright) [inline quay-tectonic-identity.md]: ... panel. Click ..., then "Save Configuration", and restart ... (note the closing quotes)

File: quay-tectonic-identity.md [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+
+From Quay Enterprise, navigate to the superuser panel, then to External Authorization, click the button: `Add OIDC Provider`, and type `tectonicidentity` in the prompt.
+
+`ODIC Server:` is the `issuer:` parameter from the tectonic-identity configmap with a forwardslash at the end: 'https://tectonic.example.com/identity/'
+
+`Client ID`: `quay-enterprise`
+
+`Client Secret:` String passed as `secret` to tectonic-identity configmap
+
+`Service Name:`	will be displayed on the login page. Suggested Value: `Tectonic Identity`
```
Comment (zbwright) [inline quay-tectonic-identity.md]: what's with the weird tab? and fix the colon to be outside the ticks, please.

File: quay-tectonic-identity.md (hunk 6 above)
Comment (zbwright) [reply quay-tectonic-identity.md]: and is this the right sequence? any reason not to grab the redirectURIs first?

File: quay-tectonic-identity.md (hunk 14 above)
Comment (kbrwn) [reply quay-tectonic-identity.md]: colon is inside `tilds` to match what is displayed in the QE ui.

File: quay-tectonic-identity.md (hunk 11 above)
Comment (kbrwn) [reply quay-tectonic-identity.md]: back to editing is informative enough IMO and was informative enough for our L2 support eng

File: quay-tectonic-identity.md (hunk 6 above)
Comment (kbrwn) [reply quay-tectonic-identity.md]: it is URI and yes the sequencing is meant to provide the most likely chance at success on first configuration by ensuring the user can copy and paste the redirect uri's from the superuser panel

File: quay-enterprise/quay-tectonic-identity.md [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+## Configure Tectonic Identity
+
+Edit the tectonic-identity configmap: 
+
+```
+kubectl -n tectonic-system edit cm tectonic-identity
+```
+
+Add an client definition under the `staticClients` section for Quay Enterprise:
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: Add a client definition ...

File: quay-enterprise/quay-tectonic-identity.md [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+kubectl -n tectonic-system get secret tectonic-ingress-tls-secret -o json | jq -r '.data["tls.crt"]' | base64 -d >> tectonic-ca-ingress.crt
+```
+
+## Upload certificate to Quay Enterprise
+
+Upload the `tectonic-ca-ingress.crt` certificate to to Quay Enterprise through the superuser panel, or follow [Adding TLS Certificates to the Quay Enterprise Container][add-certs] to upload manually.
+
+## Configure Tectonic Identity
+
+Edit the tectonic-identity configmap:
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: Edit the Tectonic Identity ConfigMap.

File: quay-enterprise/quay-tectonic-identity.md [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+  - id: quay-enterprise
+    name: Quay Enterprise
+    secret: rEpLaCeThIs
+    redirectURIs:
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback'
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback/attach'
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback/cli'
+```
+
+__secret:__ a user generated string that will later be passed to Quay Enterprise. `date | md5sum` or `openssl rand -base64 15` may be used to generate a random string.
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: ... Use `date | md5sum` or `openssl rand -base64 15` to generate a random string.

File: quay-enterprise/quay-tectonic-identity.md [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+    secret: rEpLaCeThIs
+    redirectURIs:
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback'
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback/attach'
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback/cli'
+```
+
+__secret:__ a user generated string that will later be passed to Quay Enterprise. `date | md5sum` or `openssl rand -base64 15` may be used to generate a random string. 
+
+__redirectURIs:__ callback URLs that must be passed to the tectonic-identity configmap. Replace reg.example.com with the Server Hostname of the registry. These values can be copied from the superuser panel if the oidc provider is added to Quay Enterprise before this configmap change is saved.
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: Tectonic Identity ConfigMap. Replace `reg.example.com` with the server hostname of the registry. ... OIDC provider ... ConfigMap ...

File: quay-enterprise/quay-tectonic-identity.md [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback'
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback/attach'
+      - 'https://reg.example.com/oauth2/tectonicidentity/callback/cli'
+```
+
+__secret:__ a user generated string that will later be passed to Quay Enterprise. `date | md5sum` or `openssl rand -base64 15` may be used to generate a random string. 
+
+__redirectURIs:__ callback URLs that must be passed to the tectonic-identity configmap. Replace reg.example.com with the Server Hostname of the registry. These values can be copied from the superuser panel if the oidc provider is added to Quay Enterprise before this configmap change is saved. 
+
+Save the changes to the `tectonic-identity` configmap and issue a `kubectl patch` to update the deployment:
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: Tectonic Identity ConfigMap

File: quay-enterprise/quay-tectonic-identity.md [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+__redirectURIs:__ callback URLs that must be passed to the tectonic-identity configmap. Replace reg.example.com with the Server Hostname of the registry. These values can be copied from the superuser panel if the oidc provider is added to Quay Enterprise before this configmap change is saved. 
+
+Save the changes to the `tectonic-identity` configmap and issue a `kubectl patch` to update the deployment: 
+
+```
+kubectl --namespace tectonic-system patch deployment tectonic-identity \
+    --patch "{\"spec\":{\"template\":{\"metadata\":{\"annotations\":{\"date\":\"`date +'%s'`\"}}}}}"
+```
+
+## Add an OIDC Provider for Tectonic Identity
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: OIDC provider for ...

File: quay-enterprise/quay-tectonic-identity.md [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+
+Navigate to the superuser panel, then to External Authorization, click the button: __Add OIDC Provider__, and type `tectonicidentity` in the prompt.
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: ... click "Add OIDC Provider" and type ...

File: quay-enterprise/quay-tectonic-identity.md [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+
+__OIDC Server:__ FQDN of the Tectonic cluster with `/identity/` appended. Most easily found in the tectonic-identity configmap as the `issuer:` parameter. The forwardslash at the end is important! Example: `https://tectonic.example.com/identity/`
+
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: don't include colons with parameter names ... the `issuer` parameter. 

... the `issuer` parameter. Be certain to include the forward slash at the end of the domain name. For example: ...

File: quay-enterprise/quay-tectonic-identity.md [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+__Client ID:__ `quay-enterprise`
+
+__Client Secret:__ string passed as `secret` to tectonic-identity configmap.
+
+__Service Name:__	will be displayed on the login page. Suggested Value: `Tectonic Identity`
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: ... Suggested value: ...

File: quay-enterprise/quay-tectonic-identity.md [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+
+## Configure Access Settings
+
+__Enable Open User Creation__ must be checked to allow for new users to be created. 
+
+__Enable Invite-only User Creation__ can be used to restrict access based on email invitations to organization teams. This requires an SMTP server.
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: used to restrict ...

File: quay-enterprise/quay-tectonic-identity.md [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+
+__Allow external application tokens__  if checked, users will be able to generate external application tokens for use with the docker and rkt CLI. Consider the Internal Authentication before choosing.
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: Docker and rkt CLIs. Consider the means of internal authentication before selecting this option.

File: quay-enterprise/quay-tectonic-identity.md [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+
+__Enable Open User Creation__ must be checked to allow for new users to be created. 
+
+__Enable Invite-only User Creation__ can be used to restrict access based on email invitations to organization teams. This requires an SMTP server.
+
+__Allow external application tokens__  if checked, users will be able to generate external application tokens for use with the docker and rkt CLI. Consider the Internal Authentication before choosing. 
+
+## Configure Internal Authentication
+
+__Local Database__ can be used if traditional username/password authentication with the docker cli is desired. This will require that oidc users create a password from Account Settings after initial sign up.
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: Docker CLI ... OIDC ...

File: quay-enterprise/quay-tectonic-identity.md [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+
+__External Application Token__ requires enabling `Allow external application tokens` in Access Settings. This option will  disable username/password in favor of a token generated from Account Settings. It is suggested to set an expiration date. This option provides a single sign on experience and minimizes password complication.
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: remove extra space - ... This option will disable ...
... from Account Settings. This option provides a single sign on experience, and minimizes password complication. If enabled, set an expiration date for the token.

File: quay-enterprise/quay-tectonic-identity.md [hunk 28]
```diff
@@ -0,0 +1,79 @@
...
+
+## Configure Access Settings
+
+__Enable Open User Creation__ must be checked to allow for new users to be created. 
+
+__Enable Invite-only User Creation__ can be used to restrict access based on email invitations to organization teams. This requires an SMTP server.
+
+__Allow external application tokens__  if checked, users will be able to generate external application tokens for use with the docker and rkt CLI. Consider the Internal Authentication before choosing. 
+
+## Configure Internal Authentication
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: Configure internal authentication

File: quay-enterprise/quay-tectonic-identity.md [hunk 29]
```diff
@@ -0,0 +1,79 @@
...
+
+__OIDC Server:__ FQDN of the Tectonic cluster with `/identity/` appended. Most easily found in the tectonic-identity configmap as the `issuer:` parameter. The forwardslash at the end is important! Example: `https://tectonic.example.com/identity/`
+
+__Client ID:__ `quay-enterprise`
+
+__Client Secret:__ string passed as `secret` to tectonic-identity configmap.
+
+__Service Name:__	will be displayed on the login page. Suggested Value: `Tectonic Identity`
+
+## Configure Access Settings
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: Configure access settings

File: quay-enterprise/quay-tectonic-identity.md [hunk 30]
```diff
@@ -0,0 +1,79 @@
+# Quay Enterprise OIDC Auth with Tectonic Identity 
+
+This document explains how to configure Quay Enterprise to use Tectonic Identity as an OIDC provider for authorization and authentication.
+
+## Export certificate information for TLS verification of Tectonic Identity:
```
Comment (zbwright) [inline quay-enterprise/quay-tectonic-identity.md]: remove `:`

(if these are headers, they shouldn't have colons at the end of them.)

File: quay-enterprise/quay-tectonic-identity.md (hunk 18 above)
Comment (kbrwn) [reply quay-enterprise/quay-tectonic-identity.md]: Not changing server hostname as this is to match the value in the qe super user panel.

File: quay-enterprise/quay-tectonic-identity.md (hunk 28 above)
Comment (kbrwn) [reply quay-enterprise/quay-tectonic-identity.md]: I don't agree. This is capitalized in the QE setup tool. It is not different than "Tectonic Identity" imo

File: quay-enterprise/quay-tectonic-identity.md (hunk 29 above)
Comment (kbrwn) [reply quay-enterprise/quay-tectonic-identity.md]: same as aboce

File: quay-enterprise/quay-tectonic-identity.md (hunk 28 above)
Comment (zbwright) [reply quay-enterprise/quay-tectonic-identity.md]: ok

**Pull request merged**